interactions/utils/abc/base_context_managers.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABCMeta, abstractmethod
+from typing import TypeVar
+
+_T = TypeVar("_T")
+
+__all__ = ("BaseAsyncContextManager", "BaseContextManager")
+
+
+class BaseAsyncContextManager(metaclass=ABCMeta):
+    """A base class for async iterators."""
+
+    # I don't want to make it subclass the BaseContextManager since it forces implementation of __enter__ and __exit__
+
+    @abstractmethod
+    def __init__(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __await__(self) -> None:
```

Comment:
What's your expectation of how `__await__` will be used?

Reply:
Made the abc after I made typing and didn't think about removing it